Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -1332,22 +1332,27 @@ namespace OpenInfraPlatform {
 									loops.push_back(loop);
 								}
 								
-								std::vector<carve::geom::vector<3>> firstPointloop = loops[0];
-								std::vector<carve::geom::vector<3>> secondPointLoop = loops[1];
-
-								carve::geom::vector<3> firstPoint = firstPointloop[0];
-								carve::geom::vector<3> secondPoint = firstPointloop[1];
-								carve::geom::vector<3> thirdPoint = firstPointloop[2];
-
-								carve::geom::vector<3> firstToSecond = secondPoint - firstPoint;
-								carve::geom::vector<3> firstToThird = thirdPoint - firstPoint;
-
-								carve::geom::vector<3> normalOfPlane = carve::geom::cross(firstToSecond, firstToThird);
-								normalOfPlane.normalized();
-								
+								std::vector<std::vector<carve::geom::vector<2>>> loops2D;
+								carve::geom::vector<3> normalOfPlane = GeomUtils::computePolygonNormal(loops[0]);
 
+								for (int i = 0; i < loops.size(); i++) 
+								{
+									std::vector<carve::geom::vector<3>> loop = loops[i];
+									std::vector<carve::geom::vector<2>> loop2D;
+									for (int j = 0; j < loop.size(); j++)
+									{
+										carve::geom::vector<3> point3D = loop[j];
+										carve::geom::vector<3> distance = carve::geom::VECTOR(
+											point3D.x * normalOfPlane.x,
+											point3D.y * normalOfPlane.y, 
+											point3D.z * normalOfPlane.z);

```

Comment:
Use dot product (should be available in carve library).

Reply:
dot product gives back a double, not a vector. Here I need scalar, I guess.

Reply:
Right you are, my bad! It would be better to use a matrix here in order to "project" the vector on the plane.

Reply:
Use [this help](https://stackoverflow.com/questions/9605556/how-to-project-a-point-onto-a-plane-in-3d) (great sketches!) and the already existing function in `GeomUtils`.

Reply:
Thank you!

Reply:
Resolved:
7a17b33